lib/types/src/transaction/system/tx.rs

```diff
@@ -0,0 +1,154 @@
+use alloy::consensus::transaction::{RlpEcdsaDecodableTx, RlpEcdsaEncodableTx};
+use alloy::consensus::{Transaction, Typed2718};
+use alloy::primitives::ChainId;
+use alloy::primitives::{Address, B256, Bytes, TxKind, U256};
+use alloy::rpc::types::{AccessList, SignedAuthorization};
+use alloy_rlp::{BufMut, Decodable, Encodable};
+use serde::{Deserialize, Serialize};
+
+use crate::transaction::SystemTxType;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+pub struct SystemTransaction<T: SystemTxType> {
+    pub gas_limit: u64,
```

Comment:
Add `#[serde(rename = "gas", with = "alloy::serde::quantity")]` for RPC consistency with other types